crates/bevy_winit/src/lib.rs

```diff
@@ -44,9 +46,15 @@ impl Plugin for WinitPlugin {
             .init_resource::<WinitSettings>()
             .set_runner(winit_runner)
             .add_system_to_stage(CoreStage::PostUpdate, change_window.label(ModifiesWindows));
+        #[cfg(target_arch = "wasm32")]
+        {
+            app.add_plugin(web_resize::CanvasParentResizePlugin);
+        }
         let event_loop = EventLoop::new();
-        handle_initial_window_events(&mut app.world, &event_loop);
-        app.insert_non_send_resource(event_loop);
+        let mut create_window_reader = WinitCreateWindowReader::default();
+        handle_create_window_events(&mut app.world, &event_loop, &mut create_window_reader.0);
```

Comment:
Is there any reason this has to live *here* specifically (requiring the `WinitCreateWindowReader`), rather than in `winit_runner_with`? Is it just wanting to minimise the time that the event loop exists without the window having been created?

Winit's code is nasty to read, so it's unclear if this matters. I'd be inclined to think it doesn't though.

Reply:
Yup for wasm/webgl reasons this needs to happen prior to bevy_render initialization. `winit_runner_with` executes after bevy_render init. I'll add a comment.